server/src/interfaces.ts

```diff
@@ -24,12 +24,19 @@ export interface CustomerPayload {
   address?: string;
 }
 
+/**
+ * CustomerDatastoreReponse Interface that contains the fields of the Response that
+ * server side receives from the Datastore.
+ */
+export interface CustomerDatastoreReponse extends Required<CustomerPayload> {
+  id: number;
+}
+
 /**
  * CustomerResponse Interface that contains the fields of the Response that
  * client side would receive.
  */
-export interface CustomerResponse extends Required<CustomerPayload> {
-  id: number;
+export interface CustomerResponse extends CustomerDatastoreReponse {
```

Comment:
Why `extends CustomerDatastoreResponse` not `extends Required<...>`?

Reply:
@karenfrilya97 Haha cos like all fields in CustomerDatastoreReponse is already required leh

Reply:
but `id` is not required right?

Reply:
What do you mean?  `CustomerDatastoreReponse` contains `id` as a non-optional field.
`Required<T>` is a TypeScript utility type that that makes all fields of type T required (aka non-optional). 
For example CustomerPayload includes optional fields
```
export interface CustomerPayload {
  gpayId: string;
  contactNumber?: string; // E164 format
  address?: string;
}
```
When I did 
```
export interface CustomerResponse extends Required<CustomerPayload> {
  id: number;
  numOngoingCommits: number;
}
```
It means `CustomerResponse` interface contains `contactNumber` and `address` as non-optional fields.

Reply:
@hzxcaryn Okay never mind, I misunderstood the `Required<T>` utility type hahahahah :/ Thanks for explaining! Because Typescript doesn't enforce that the object contains the non-optional fields during runtime then I thought `Required<T>` would do that. But it doesn't. So you are right. Hahahah sorry for making you explain all this! But thanks for explaining anyway!! :D